src/btree/bt_curnext.c

```diff
@@ -732,13 +749,29 @@ __wt_btcur_next_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
     WT_SESSION_IMPL *session;
     size_t total_skipped, skipped;
     uint32_t flags;
-    bool newpage, prefix_key_out_of_bounds, restart;
+    bool key_out_of_bounds, newpage, restart, need_walk;
 
     cursor = &cbt->iface;
-    prefix_key_out_of_bounds = false;
+    key_out_of_bounds = false;
+    need_walk = false;
     session = CUR2S(cbt);
     total_skipped = 0;
 
+    /*
```

Comment:
ideally we are not supposed to have this code segment before __wt_cursor_func_init in line 783. Is there a reason why this has to be here itself ?

Reply:
After some discussion and trials, we decided to maintain the cursor positioning logic above `__wt_cursor_func_init` for various reasons:

- Moving this logic below the `WT_ERR(__wt_cursor_func_init(cbt, false))` call results in a segfault. As part of `__wt_cursor_func_init`, the WT_CBT_ACTIVE flag is set to signal that the page is active. This is an incorrect assumption in our case of positioning the cursor, and will lead to checking if the current page is pinned, accessing invalid memory (cbt->ref->page) , and resulting in a segfault.

- As part of our cursor positioning logic, `__wt_btcur_search_near` is called, and that will already call `__wt_cursor_func_init` .

- As part of positioning, we call search_near to position the cursor according to the bound. `search_near calls` -> `__wt_cursor_func_init` -> `__wt_txn_cursor_op` that takes a snapshot if the current transaction doesn’t have one existing already. Once search_near has taken the snapshot, it doesn’t clear it. This snapshot will be the same one used throughout the rest of the function (`__wt_btcur_next_prefix`) as the next time the `_wt_cursor_func_init` -> `__wt_txn_cursor_op` code flow is hit, the WT_TXN_HAS_SNAPSHOT flag will already have been set, and no new snapshot will be taken. As the snapshot used throughout is consistent, this will not lead to any data visibility issues.